File: rt-master/src/main/java/xyz/redtorch/node/db/ZookeeperService.java
package xyz.redtorch.node.db;

import com.alibaba.fastjson.JSONObject;
import xyz.redtorch.common.util.ZookeeperUtils;

import java.util.List;

public interface ZookeeperService {

    ZookeeperUtils getZkUtils();

    List<JSONObject> find(String collection);

    JSONObject findById(String collection, String id);

    boolean update(String collection, String id, String data);

    boolean insert(String collection, String id, String data);

    boolean upsert(String collection, String id, String data);

    boolean deleteById(String collection, String id);

}

File: rt-master/src/main/kotlin/xyz/redtorch/master/db/ZookeeperService.kt
package xyz.redtorch.master.db

import com.fasterxml.jackson.databind.JsonNode
import xyz.redtorch.common.utils.ZookeeperUtils

interface ZookeeperService {
    fun getZkUtils(): ZookeeperUtils

    fun get3DesKey(): String

    fun find(collection: String, enableEncryption: Boolean = true): List<JsonNode>?

    fun findById(collection: String, id: String, enableEncryption: Boolean = true): JsonNode?

    fun update(collection: String, id: String, data: String, enableEncryption: Boolean = true): Boolean

    fun insert(collection: String, id: String, data: String, enableEncryption: Boolean = true): Boolean

    fun upsert(collection: String, id: String, data: String, enableEncryption: Boolean = true): Boolean

    fun deleteById(collection: String, id: String): Boolean
}